src/scraper/scraper.cpp

```diff
@@ -2497,6 +2404,72 @@ bool MarkScraperFileManifestEntryNonCurrent(ScraperFileManifestEntry& entry)
+    {
+        LOCK(cs_StructScraperFileManifest);
+        if (fDebug3) _log(logattribute::INFO, "LOCK", "download "+ filetype + " files: AlignScraperFileManifestEntries: cs_StructScraperFileManifest");
+
+        // Iterate mScraperFileManifest to find any prior filetype records for the same project and change current flag to false,
+        // or delete if older than SCRAPER_FILE_RETENTION_TIME or non-current and fScraperRetainNonCurrentFiles
+        // is false.
+
+        ScraperFileManifestMap::iterator entry;
+        for (entry = StructScraperFileManifest.mScraperFileManifest.begin(); entry != StructScraperFileManifest.mScraperFileManifest.end(); )
+        {
+            ScraperFileManifestMap::iterator entry_copy = entry++;
+
+            if (entry_copy->second.project == sProject && entry_copy->second.current == true && entry_copy->second.filetype == filetype)
+            {
+                _log(logattribute::INFO, "AlignScraperFileManifestEntries", "Marking old project manifest "+ filetype + " entry as current = false.");
+                MarkScraperFileManifestEntryNonCurrent(entry_copy->second);
+            }
+
+            // If filetype records are older than EXPLORER_EXTENDED_FILE_RETENTION_TIME delete record, or if fScraperRetainNonCurrentFiles is false,
+            // delete all non-current records, including the one just marked non-current. (EXPLORER_EXTENDED_FILE_RETENTION_TIME rather
+            // then SCRAPER_FILE_RETENTION_TIME is used, because this section is only active if fExplorer is true.)
+            if (entry_copy->second.filetype == filetype && (((GetAdjustedTime() - entry_copy->second.timestamp) > EXPLORER_EXTENDED_FILE_RETENTION_TIME)
+                                                          || (entry_copy->second.project == sProject && entry_copy->second.current == false && !SCRAPER_RETAIN_NONCURRENT_FILES)))
+            {
+                DeleteScraperFileManifestEntry(entry_copy->second);
+            }
+        }
+
+        if (!InsertScraperFileManifestEntry(NewRecord))
+            _log(logattribute::WARNING, "AlignScraperFileManifestEntries", "Manifest entry already exists for " + NewRecord.hash.ToString() + " " + file_name);
+        else
+            _log(logattribute::INFO, "AlignScraperFileManifestEntries", "Created manifest entry for " + NewRecord.hash.ToString() + " " + file_name);
+
+        // The below is not an ideal implementation, because the entire map is going to be written out to disk each time.
+        // The manifest file is actually very small though, and this primitive implementation will suffice. I could
+        // put it up in the while loop above, but then there is a much higher risk that the manifest file could be out of
+        // sync if the wallet is ended during the middle of pulling the files.
+        _log(logattribute::INFO, "DownloadProjectTeamFiles", "Persisting manifest entry to disk.");
```

Comment:
It happened again 😄 

Sorry to be a drag...

Reply:
That is what reviews are for...

Reply:
It is called distracted coding. Lucky I can get any of this done I have so much going on.

Reply:
It amazes me that you do.

Reply:
Ok. I think I got it fixed properly this time. Also I had forgotten the code block in the processed one, and added the exclusion flag as a parameter to the function to handle that.